web/src/message_list_data.js

```diff
@@ -223,7 +223,11 @@ export class MessageListData {
                 return true;
             }
             return (
-                !user_topics.is_topic_muted(message.stream_id, message.topic) || message.mentioned
+                !user_topics.topic_has_visibility_policy(
+                    message.stream_id,
+                    message.topic,
+                    user_topics.visibility_policy.MUTED,
+                ) || message.mentioned
```

Comment:
Reading this commit, I think we actually want to keep a `user_topics.is_topic_muted` function that is a thin wrapper around `user_topics.topic_has_visibility_policy`; each of these blocks is way harder to read than the previous code.